interop-cats/shared/src/main/scala/zio/interop/queue/CQueue.scala

```diff
@@ -0,0 +1,72 @@
+package zio.interop.queue
+
+import cats.effect.{ Effect, LiftIO }
+import zio.RIO
+import zio.Runtime
+import zio.interop.queue.CQueue._
+
+final class CQueue[F[+_], -A, +B] private[queue] (private val q: CompatQueue[A, B]) {
+
+  def awaitShutdown(implicit R: Runtime[Any], F: LiftIO[F]): F[Unit] = toEffect(q.awaitShutdown)
+
+  def capacity: Int = q.capacity
+
+  def isShutdown(implicit R: Runtime[Any], F: LiftIO[F]): F[Boolean] = toEffect(q.isShutdown)
+
+  def offer(a: A)(implicit R: Runtime[Any], F: LiftIO[F]): F[Boolean] = toEffect(q.offer(a))
+
+  def offerAll(as: Iterable[A])(implicit R: Runtime[Any], F: LiftIO[F]): F[Boolean] = toEffect(q.offerAll(as))
+
+  def shutdown(implicit R: Runtime[Any], F: LiftIO[F]): F[Unit] = toEffect(q.shutdown)
+
+  def size(implicit R: Runtime[Any], F: LiftIO[F]): F[Int] = toEffect(q.size)
+
+  def take(implicit R: Runtime[Any], F: LiftIO[F]): F[B] = toEffect(q.take)
+
+  def takeAll(implicit R: Runtime[Any], F: LiftIO[F]): F[List[B]] = toEffect(q.takeAll)
+
+  def takeUpTo(max: Int)(implicit R: Runtime[Any], F: LiftIO[F]): F[List[B]] = toEffect(q.takeUpTo(max))
+
+  def &&[A0 <: A, C](that: CQueue[F, A0, C]): CQueue[F, A0, (B, C)] = new CQueue(q && that.q)
+
+  def both[A0 <: A, C](that: CQueue[F, A0, C]): CQueue[F, A0, (B, C)] = new CQueue(q.both(that.q))
+
+  def bothWith[A0 <: A, C, D](that: CQueue[F, A0, C])(f: (B, C) => D): CQueue[F, A0, D] =
+    new CQueue(q.bothWith(that.q)(f))
+
+  def bothWithM[A0 <: A, C, D](
+    that: CQueue[F, A0, C]
+  )(f: (B, C) => F[D])(implicit R: Runtime[Any], E: Effect[F]): CQueue[F, A0, D] =
+    new CQueue(q.bothWithM(that.q) { (b, c) =>
+      fromEffect(f(b, c))
+    })
+
+  def contramap[C](f: C => A): CQueue[F, C, B] = new CQueue(q.contramap(f))
+
+  def contramapM[C](f: C => F[A])(implicit R: Runtime[Any], E: Effect[F]): CQueue[F, C, B] =
+    new CQueue(q.contramapM(c => fromEffect(f(c))))
+
+  def filterInput[A0 <: A](f: A0 => Boolean): CQueue[F, A0, B] = new CQueue(q.filterInput(f))
+
+  def filterInputM[A0 <: A](f: A0 => F[Boolean])(implicit R: Runtime[Any], E: Effect[F]) =
+    new CQueue(q.filterInputM((a0: A0) => fromEffect(f(a0))))
+
+  def map[C](f: B => C): CQueue[F, A, C] = new CQueue(q.map(f))
+
+  def mapM[C](f: B => F[C])(implicit R: Runtime[Any], E: Effect[F]) = new CQueue(q.mapM(b => fromEffect(f(b))))
+
+  def poll(implicit R: Runtime[Any], F: LiftIO[F]): F[Option[B]] = toEffect(q.poll)
+}
+
+object CQueue {
+  import zio.interop.catz._
+  import zio.Runtime
+
+  private type CompatQueue[-A, +B] = zio.ZQueue[Any, Throwable, Any, Throwable, A, B]
```

Comment:
We are only using this type alias one place so we can probably just inline it.

Reply:
done 👍